### DISCUSSION TITLE
What is the best way to read the request body in a Play filter?

### DISCUSSION THREAD

**Discussion by tanin47** [playframework/playframework#13010] [Q&A] [answered]
I have some use cases where I need to read the request body. It seems the filter only gets RequestHeader.

https://github.com/orgs/playframework/discussions/12656 shows how to do it but the code is fairly long and complex. 

I wonder if there's a better way.

**Comment by mkurz** [comment, marked as answer]
Did you solve your need? I think you should continue the discussion with the author in
- #12656

he might be able to help you.
